### PR TITLE
remove member from cache when ban

### PR DESCRIPTION
**Pull request by PyGaVS**
### What are the changes?
Removing the member from the server cache : 
`marshaller.cache.remove(memberCacheKey)`
And updating the server in the cache :
`marshaller.cache.put(serverCacheKey, rawServer)`

### How to test the changes?
Replace that :
```dart
await Future.wait([
      marshaller.cache.put(serverCacheKey, rawServer),
      marshaller.cache.put(channelCacheKey, rawChannel)
    ]);
```

By that :
```dart
marshaller.cache.has(memberCacheKey).then((value) => print(value));
    await Future.wait([
      marshaller.cache.remove(memberCacheKey),
      marshaller.cache.put(serverCacheKey, rawServer),
    ]).then((value) => marshaller.cache.has(memberCacheKey).then((value) => print(value)));
```

It should log true at the beginning when the member is not removed from the cache and false and the end when the member is removed.

### CODE REVIEW

File: lib/infrastructure/internals/packets/listeners/guild_ban_add_packet.dart [hunk 1]
```diff
@@ -19,11 +19,21 @@ final class GuildBanAddPacket implements ListenablePacket {
...
     final member = server.members.list[user.id];
+    final memberId = message.payload['user']['id'];
 
-    logger.trace('GuildBanAddPacket: ${user.username} was ban in ${server.name}');
+    final memberCacheKey = marshaller.cacheKey.serverMember(serverId: server.id, memberId: memberId);
+    final rawServer = await marshaller.serializers.server.deserialize(server);
+    final serverCacheKey = marshaller.cacheKey.server(server.id);
 
-    dispatch(event: Event.serverBanAdd, params: [member, user, server]);
+    logger.trace('GuildBanAddPacket: ${user.username} was ban in ${server.name}');
```
Comment (LeadcodeDev) [inline lib/infrastructure/internals/packets/listeners/guild_ban_add_packet.dart:28]: You can remove this line

File: lib/.DS_Store
Comment (LeadcodeDev) [inline lib/.DS_Store:1]: Please remove this file

File: .DS_Store
Comment (LeadcodeDev) [inline .DS_Store:1]: Please remove this file

**Review by LeadcodeDev — CHANGES_REQUESTED** [review]

File: lib/infrastructure/internals/packets/listeners/guild_ban_add_packet.dart (hunk 1 above)
Comment (PyGaVS) [reply lib/infrastructure/internals/packets/listeners/guild_ban_add_packet.dart:28]: ok

**Review by LeadcodeDev — APPROVED** [review]

**Pull request merged**